### ISSUE TITLE
Is help needed to upgrade this library to not require polling?

### ISSUE THREAD

**Issue by kylebakerio** [DougReeder/aframe-button-controls#3]
I find the 'poll' option confusing/strange. Is the problem fundamental to webxr spec, or is it just a matter of making the code easier to write?

Would borrowing code from [this project](https://github.com/TakashiYoshinaga/Oculus-Quest-Input-Sample) help?

I'm using this library in a project I'm working, was considering contributing an upstream patch if it would improve the code.

Cheers.

**Comment by DougReeder** [comment]
The only WebXR button events are select and squeeze (and squeeze wasn't implemented initially).  To retain the same behavior as under WebVR (firing an event for **any** button) polling is necessary.

New projects should **not** use polling, and require the user to use the select or squeeze buttons.

**Comment by kylebakerio** [comment]
I use this project with `poll=true` to catch "all" buttons, because I needed all buttons. Perhaps I'm not using the library completely as designed? To me it's just a simple interface to catch any button as an event in an easy way--I assign different values to each button, so just using only select and squeeze is not ideal.

Perhaps there's a better tool for the job or an easier route out there, I just picked this one at the time because I didn't see any way to get the cardboard button working besides this library, so I just put this library at the middle of how I interface with all controllers.

(Ironically, to get the cardboard button working while poll=true, which I needed for my project, I had to dig deep and figure it all out anyways, but I just saw it as a shortcut at the time.)

**Comment by kylebakerio** [comment]
Is there a reason that using `poll=true` is bad? That's how the old school gamepad API works, based on this library, I just assumed that was how the webxr full controller spec worked.

**Comment by DougReeder** [comment]
With the change from the WebVR proposal to the WebXR standard came a change in paradigm. The new paradigm (select & squeeze events) allows better time resolution, which matters when throwing an object.  It also standardizes primary and secondary buttons, for a more consistent user experience.

aframe-button-controls will continue to support polling indefinitely, for the projects that were built with the assumption that all buttons should do the same thing.

**Comment by kylebakerio** [comment]
I see. So, this library just isn't interested in catching 'all buttons' when poll !== true? It just assumes it is enough to catch squeeze and select? (I haven't tried that, but that's how it seems to me when I try to reason about how it must work now?) I use polling in this library to catch the individual buttons and use them each independently--iirc, off the top of my head now, I thought the readme said that if you want to use all buttons, you have to use poll=true, but I feel the need to go back through the readme or something.

I was just confused this whole time that the 'new' spec (don't use polling) resulted in decreased functionality (can't use all buttons, only select/squeeze). I therefore assumed the newer spec was the polling spec. Interesting, though.

Outside of this library, I haven't looked at all how buttons work in the modern webxr spec.

**Comment by DougReeder** [comment]
Under WebXR, with poll=false, only the buttons for select and squeeze are detected. For an unfamiliar webapp, users don't necearily expect all the buttons to do something.  

Poll=true  exists mostly so old webapps don't lose functionality.

**Comment by kylebakerio** [comment]
That's true, and for the vast majority of webapps, that totally makes sense.

I explicitly don't make them do anything by default, but that means a full menu system for a complex app. I went ahead and did that, but also created an interactive key-assigning menu interface where menu functions can be assigned to buttons by the user as desired, and their profiles saved to localStorage.

I had intended to make it available to other users, but for a number of reasons, that ended up not being generalizable in my first draft, and will later require a rewrite to break it out if I want to share it. But in any case, this library, along with poll=true, was invaluable. As far as I can tell reading the webxr spec, the polling functionality is not deprecated I think, no?

But I get what you're saying, it seems your vision for this library is really to just support the very basic controller actions, and nothing more. It seems I may be using it a bit beyond its intended function here? Not sure if I should just fork it.

As for the pull request, I have my fork working for me locally, I'll try to give your refactored version a try soon and see if it works for me or not and go from there.
